src/m365/outlook/commands/message/message-remove.ts

```diff
@@ -0,0 +1,148 @@
+    return commands.MESSAGE_REMOVE;
+  }
+
+  public get description(): string {
+    return 'Removes a specific message from a mailbox';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+    this.#initTypes();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        userId: typeof args.options.userId !== 'undefined',
+        userName: typeof args.options.userName !== 'undefined',
+        force: !!args.options.force
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-i, --id <id>'
+      },
+      {
+        option: '--userId [userId]'
+      },
+      {
+        option: '--userName [userName]'
+      },
+      {
+        option: '-f, --force'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.userId && !validation.isValidGuid(args.options.userId)) {
+          return `Value '${args.options.userId}' is not a valid GUID for option 'userId'.`;
+        }
+
+        if (args.options.userName && !validation.isValidUserPrincipalName(args.options.userName)) {
+          return `Value '${args.options.userName}' is not a valid user principal name for option 'userName'.`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initTypes(): void {
+    this.types.string.push('id', 'userId', 'userName');
+    this.types.boolean.push('force');
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    const isAppOnlyAccessToken: boolean | undefined = accessToken.isAppOnlyAccessToken(auth.connection.accessTokens[this.resource].accessToken);
+    let principalUrl = '';
+
+    if (isAppOnlyAccessToken) {
+      if (!args.options.userId && !args.options.userName) {
+        throw `The option 'userId' or 'userName' is required when removing a message using application permissions.`;
+      }
+
+      if (args.options.userId && args.options.userName) {
+        throw `Both options 'userId' and 'userName' cannot be used together when removing a message using application permissions.`;
+      }
+    }
+    else {
+      if (args.options.userId && args.options.userName) {
+        throw `Both options 'userId' and 'userName' cannot be used together when removing a message using delegated permissions.`;
+      }
+    }
+
+    if (args.options.userId || args.options.userName) {
+      principalUrl += `users/${args.options.userId || formatting.encodeQueryParameter(args.options.userName!)}`;
+    }
+    else {
+      principalUrl += 'me';
+    }
+
+    const removeMessage = async (): Promise<void> => {
+      try {
+        if (this.verbose) {
+          await logger.logToStderr(`Removing message with 'id' ${args.options.id} using ${isAppOnlyAccessToken ? 'application' : 'delegated'} permissions.`);
```

Comment:
Seems like my last remark was not so clear, sorry for that!

```suggestion
          await logger.logToStderr(`Removing message with id '${args.options.id}' using ${isAppOnlyAccessToken ? 'application' : 'delegated'} permissions.`);
```

Reply:
my bad